### PR TITLE
Cf 33

### PR DESCRIPTION
**Pull request by dinocom33**
This PR represents Card details view. It provides all cards list or card details by card id.

I am attaching how the card in set details looks like:

```
{
  "id": 500,
  "yugioh_card": {
    "id": 58,
    "card_name": "Super Polymerization",
    "type": "Spell Card",
    "frame_type": "spell",
    "description": "Discard 1 card; Fusion Summon 1 Fusion Monster from your Extra Deck, using monsters from either field as Fusion Material. Neither player can activate cards or effects in response to this card's activation.",
    "attack": "NULL",
    "defense": "NULL",
    "level": "NULL",
    "race": "Quick-Play",
    "attribute": "NULL",
    "archetype": "NULL",
    "image": "https://images.ygoprodeck.com/images/cards/48130397.jpg"
  },
  "set": {
    "card_set_name": "Maximum Gold",
    "set_code": "MAGO"
  },
  "rarity": {
    "rarity": "Premium Gold Rare",
    "rarity_code": "(PG)"
  }
}
```

### CODE REVIEW

**Comment by julkascript** [comment]
The idea is to obtain the **data of the card by CardInSet id**. Follow the logic:
1. The user will type in the search the name of the card they desire to open (e.g. Dark Magician)
2. They will be presented with all the Dark Magician cards in all the sets.
3. They will choose their desired card in that set, therefore we fetch the CardInSet id.

The response should contain all YugiohCardInSet values, all YugiohCard values and the name of the card.

**Comment by dinocom33** [comment]
Added yugioh card in set endpoint which allows to view list of all cards in set or find card in set by ID

File: backend/yugioh/views.py [hunk 1]
```diff
@@ -1,3 +1,38 @@
...
+from rest_framework import permissions
 
-# Create your views here.
+from .models import YugiohCard, YugiohCardInSet
+from .serializers import YugiohSerializer, YugiohCardInSetSerializer
+from .filters import YugiohFilter
+
+
+@extend_schema(tags=['Yu-Gi-Oh Card'])
+class YugiohViewSet(viewsets.ModelViewSet):
```
Comment (julkascript) [inline backend/yugioh/views.py:13]: - [ ] Rename the viewset to `YugiohCardViewSet`

I am trying to access an item from this viewset (e.g. `/api/yugioh/cards/1/`), but I get error:
`Field 'id' expected a number but got OrderedDict([('rarity', 'Secret Rare'), ('rarity_code', '(ScR)')]).`

- [ ] Fix YugiohCard viewset access error

File: backend/yugioh/views.py [hunk 2]
```diff
@@ -1,3 +1,38 @@
-from django.shortcuts import render
+from django.db.models import Q
```
Comment (julkascript) [inline backend/yugioh/views.py:1]: Remove unused imports

File: backend/yugioh/serializers.py [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+            return None
+
+        all_sets_ids = [set_name['set'] for set_name in all_sets]
+        set_names = [
+            f'{YugiohCardSet.objects.get(id=set_id).card_set_name}({YugiohCardSet.objects.get(id=set_id).set_code})' for
+            set_id in all_sets_ids]
+        return set_names
+
+
+class YugiohCardInSetCardSerializer(YugiohSerializer):
```
Comment (julkascript) [inline backend/yugioh/serializers.py:73]: Include the `id` of the rarity and the `id` of the set in the response.

File: backend/yugioh/serializers.py [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+            'defense',
+            'level',
+            'race',
+            'attribute',
+            'archetype',
+            'image',
+        ]
+
+
+class YugiohCardInSetSerializer(serializers.ModelSerializer):
```
Comment (julkascript) [inline backend/yugioh/serializers.py:92]: Include the id of the rarity and the id of the set in the response.

File: backend/yugioh/serializers.py [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+
+
+class YugiohCardSetSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = YugiohCardSet
+        fields = ['card_set_name', 'set_code']
+        read_only_fields = ['card_set_name', 'set_code']
+
+
+class YugiohSerializer(serializers.ModelSerializer):
```
Comment (julkascript) [inline backend/yugioh/serializers.py:19]: - [ ] Rename to `YugiohCardSerializer`

File: backend/yugioh/serializers.py [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+class YugiohCardRaritySerializer(serializers.ModelSerializer):
+    class Meta:
+        model = YugiohCardRarity
+        fields = ['rarity', 'rarity_code']
+
+
+class YugiohCardSetSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = YugiohCardSet
+        fields = ['card_set_name', 'set_code']
```
Comment (julkascript) [inline backend/yugioh/serializers.py:15]: - [ ] Also include the `id`

File: backend/yugioh/views.py (hunk 2 above)
Comment (dinocom33) [reply backend/yugioh/views.py:1]: Unused imports are removed.

File: backend/yugioh/views.py (hunk 1 above)
Comment (dinocom33) [reply backend/yugioh/views.py:13]: YugiohCard viewset access error are fixed.
Тhe answer now looks like this оне:

```
{
  "id": 50,
  "card_name": "Red-Eyes Dark Dragoon",
  "type": "Fusion Monster",
  "frame_type": "fusion",
  "description": "\"Dark Magician\" + \"Red-Eyes Black Dragon\" or 1 Dragon Effect Monster\r\nCannot be destroyed by card effects. Neither player can target this card with card effects. During your Main Phase: You can destroy 1 monster your opponent controls, and if you do, inflict damage to your opponent equal to that monster's original ATK. You can use this effect a number of times per turn up to the number of Normal Monsters used as Fusion Material for this card. Once per turn, when a card or effect is activated (Quick Effect): You can discard 1 card; negate the activation, and if you do, destroy that card, and if you do that, this card gains 1000 ATK.",
  "attack": "3000",
  "defense": "2500",
  "level": "8",
  "race": "Spellcaster",
  "attribute": "DARK",
  "archetype": "Red-Eyes",
  "rarity": [
    {
      "id": 4,
      "rarity": "Ultra Rare",
      "rarity_code": "(UR)"
    },
    {
      "id": 10,
      "rarity": "Prismatic Secret Rare",
      "rarity_code": "(PScR)"
    },
    {
      "id": 2,
      "rarity": "Starlight Rare",
      "rarity_code": "(StR)"
    }
  ],
  "set": [
    {
      "id": 14,
      "card_set_name": "2020 Tin of Lost Memories Mega Pack",
      "set_code": "MP20"
    },
    {
      "id": 17,
      "card_set_name": "2022 Tin of the Pharaoh's Gods",
      "set_code": "MP22"
    },
    {
      "id": 118,
      "card_set_name": "Brothers of Legend",
      "set_code": "BROL"
    }
  ],
  "image": "https://images.ygoprodeck.com/images/cards/37818794.jpg"
}
```

File: backend/yugioh/serializers.py (hunk 4 above)
Comment (dinocom33) [reply backend/yugioh/serializers.py:92]: IDs of the rarity and set are included now.
The response looks like this one:

```
{
  "id": 50,
  "yugioh_card": {
    "id": 8,
    "card_name": "Cosmic Cyclone",
    "type": "Spell Card",
    "frame_type": "spell",
    "description": "Pay 1000 LP, then target 1 Spell/Trap on the field; banish it.",
    "attack": "NULL",
    "defense": "NULL",
    "level": "NULL",
    "race": "Quick-Play",
    "attribute": "NULL",
    "archetype": "NULL",
    "image": "https://images.ygoprodeck.com/images/cards/8267140.jpg"
  },
  "set": {
    "id": 6,
    "card_set_name": "Duel Devastator",
    "set_code": "DUDE"
  },
  "rarity": {
    "id": 4,
    "rarity": "Ultra Rare",
    "rarity_code": "(UR)"
  }
}
```

**Review by julkascript — APPROVED** [review]

**Pull request merged**